### PR TITLE
Refactoring the timeline

### PR DESCRIPTION
**Pull request by jaril**
This PR doesn't change anything functionally. It just cleans up the one timeline index file so that it's easier to work on the timeline.

Stress test/Demo: https://share.descript.com/view/UyoJEcPOFqD

### CODE REVIEW

**Review by jasonLaster — APPROVED** [review]
Looks great and works well!

File: src/ui/actions/timeline.ts [hunk 1]
```diff
@@ -176,3 +180,175 @@ export function seek(
...
+    let nextGraphicsPromise = getGraphicsAtTime(nextGraphicsTime);
+
+    const prepareNextGraphics = () => {
+      nextGraphicsTime = nextPaintOrMouseEvent(currentTime)?.time || endTime;
+      nextGraphicsPromise = getGraphicsAtTime(nextGraphicsTime);
+    };
+    const shouldContinuePlayback = () => selectors.getPlayback(getState());
+    prepareNextGraphics();
+
+    while (shouldContinuePlayback()) {
```
[the remark this answers is not in the document]
Comment (jasonLaster) [reply src/ui/actions/timeline.ts:261]: meh, i can go either way... it is used 3 times and it is nice to describe the intent

[the remark this answers is not in the document]
Comment (jaril) [reply src/ui/actions/timeline.ts:261]: Yeah didn't want to keep repeating it. I'll keep it for readability.

File: src/ui/actions/timeline.ts [hunk 2]
```diff
@@ -176,3 +180,175 @@ export function seek(
...
+        return;
+      }
+
+      currentDate = Date.now();
+      currentTime = startTime + (currentDate - startDate);
+
+      if (currentTime > endTime) {
+        log(`FinishPlayback`);
+        seekToTime(endTime);
+        dispatch(setTimelineState({ currentTime: endTime, playback: null }));
```
[the remark this answers is not in the document]
Comment (jaril) [reply src/ui/actions/timeline.ts]: Fixed!

**Pull request merged**